weatherman.py

```diff
@@ -0,0 +1,80 @@
+"""
+Generating reports of temperature from given text files.
+"""
+
+import os
+import argparse
+import sys
+
+file_list = os.listdir('./weatherdata/')
+hottest_day = {}
+hottest_date = {}
+coldest_day = {}
+maxi_humidity= {}
+mini_humidity= {}
+
+parser = argparse.ArgumentParser(prog='weatherman'
+								, usage='%(prog)s [report#] [data_dir]'
+								, description='Generate weather reports')
+parser.add_argument('-d','--data_dir', type = str, metavar = ''
+					, help = 'Directory containing weather data files'
+					, required=True)
+parser.add_argument('-r','--report_number', type = str, metavar = ''
+					, choices = ['1', '2'], help = '1 for Annual Max/Min Temperature, 2 for Hottest day of each year'
+					, required=True)
+try:
+    args = parser.parse_args()	
+except:
```

Comment:
`parse_args()` does not raise exceptions.

Reply:
Used this to print the help/usage info if wrong or no argument is provided. This was also asked to do in the question.
I'll take it out if it is not required or do let me know if there is a better way of doing this.

Reply:
parse_args will print help/usage if wrong arguments are provided. `parser.print_help()` will not execute as exception will not occur.

Reply:
I have a bit of a different view on this, but you are right, the required help is being printed.
I took it out.